### PR TITLE
Add filtering for reconciliation collection view

### PR DESCRIPTION
**Pull request by teneightfive**
This adds support for the macros form builder to display filters to
the side of the results as well as changing the middleware and
controllers to support sending the request data to the API.

## What it looks like

localhost_3001_omis_reconciliation_sortby created_on 3adesc status quote_accepted

### CODE REVIEW

File: src/apps/omis/apps/list/controllers.js [hunk 1]
```diff
@@ -21,7 +21,22 @@ function renderCollectionList (req, res) {
 }
 
 function renderReconciliationList (req, res) {
-  res.render('omis/apps/list/views/list-reconciliation')
+  const sortForm = merge({}, collectionSortForm, {
+    hiddenFields: Object.assign({}, omit(req.query, 'sortby')),
```
Comment (yeahfro) [inline src/apps/omis/apps/list/controllers.js]: `assign`

File: src/apps/omis/apps/list/controllers.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const { assign, merge, omit } = require('lodash')
 
 const { buildSelectedFiltersSummary } = require('../../../builders')
-const { collectionFiltersFields, collectionSortForm } = require('./macros')
+const { collectionFiltersFields, reconciliationFiltersFields, collectionSortForm } = require('./macros')
```
Comment (feedmypixel) [inline src/apps/omis/apps/list/controllers.js]: separate line?

File: test/unit/apps/omis/apps/list/middleware.test.js [hunk 3]
```diff
@@ -44,6 +44,26 @@ describe('OMIS list middleware', () => {
     })
   })
 
+  describe('#setReconciliationResults', () => {
+    beforeEach(async () => {
+      this.req.query = {
+        status: 'draft',
+        company_name: 'samsung',
```
Comment (feedmypixel) [inline test/unit/apps/omis/apps/list/middleware.test.js]: can you change this to a non company?

Comment (teneightfive) [reply test/unit/apps/omis/apps/list/middleware.test.js]: What's the reason for that?

**Pull request merged**